python/src/spark_rapids_ml/umap.py

```diff
@@ -507,19 +526,67 @@ def raw_data(self) -> List[List[float]]:
     def _get_cuml_transform_func(
         self, dataset: DataFrame, category: str = transform_evaluate.transform
     ) -> Tuple[_ConstructFunc, _TransformFunc, Optional[_EvaluateFunc],]:
+        cuml_alg_params = self.cuml_params.copy()
+        driver_embedding = np.array(self.embedding_, dtype=np.float32)
```

Comment:
maybe the broadcast way can be more efficient since we don't need to package the embedding or raw_data to every task and send it to executor from driver